zict/file.py

```diff
@@ -123,13 +128,9 @@ def __setitem__(
         | list[bytes | bytearray | memoryview]
         | tuple[bytes | bytearray | memoryview, ...],
     ) -> None:
-        try:
-            del self[key]
-        except KeyError:
-            pass
-
+        self.discard(key)
         fn = self._safe_key(key)
-        with open(os.path.join(self.directory, fn), "wb") as fh:
+        with open(os.path.join(self.directory, fn), "wb") as fh, self.unlock():
```

Comment:
Curious why writing files is considered 'safe' to release the lock but deleting it is not (see __delitem__)? Seems mutation of files ought to (probably) be locked, or consistently unlocked and left to user to worry about.

Reply:
We are not mutating files though. We're always creating a new one. If two threads call `__setitem__` on the same key, they'll end up writing to two different files.
I just noticed that I forgot about a race condition there though - one of the two files would remain there indefinitely, littering the hard drive. Fixed.

It's not that deleting files is unsafe - it's that `os.remove` is expected to be _fast_ (see definition above) so I chose to keep the method simple. For the record, `dict.pop` (first line of `__delete__`) is *not* thread-safe and would need to be wrapped in a lock.